src/core/stream/representation/check_for_discontinuity.ts

```diff
@@ -78,105 +81,141 @@ export default function checkForDiscontinuity(
   }
 
   if (nextBufferedInRangeIdx === undefined) {
-    const discontinuityEnd = representation.index.checkDiscontinuity(checkedRange.start);
-    if (discontinuityEnd !== null) {
-      return { start: undefined,
-               end: discontinuityEnd };
+    // There's no segment currently buffered for the current range.
+
+    if (nextSegmentStart === null) { // No segment to load in that range
+      // Check if we are in a discontinuity at the end of the current Period
+      if (hasFinishedLoading &&
+          period.end !== undefined &&
+          checkedRange.end >= period.end)
+      {
+        return { start: undefined, end: null }; // discontinuity to Period's end
+      }
+
+      // Check that there is a discontinuity anounced in the Manifest there
```

Comment:
announced*

Reply:
done (as well as all other mentions of "anounce" in the code)